quint/src/IRprinting.ts

```diff
@@ -35,8 +37,16 @@ export function moduleToString(quintModule: QuintModule): string {
  *
  * @returns a string with the pretty printed definition
  */
-export function definitionToString(def: QuintDef, includeBody: boolean = true): string {
-  const typeAnnotation = isAnnotatedDef(def) ? `: ${typeToString(def.typeAnnotation)}` : ''
+export function definitionToString(
+  def: QuintDef,
+  includeBody: boolean = true,
+  type: TypeScheme | undefined = undefined
+): string {
+  const typeAnnotation = isAnnotatedDef(def)
+    ? `: ${typeToString(def.typeAnnotation)}`
+    : type
+    ? `: ${typeSchemeToString(type)}`
+    : ''
```

Comment:
Could you add some parentheses here? I am not sure about the priorities

Reply:
Actually, I can't. `prettier` removes it. I've added a comment to clarify priorities instead.